core/modules/node/node.module

```diff
@@ -1888,6 +1886,34 @@ function node_block_view($delta = '', $settings = array()) {
         }
       }
       break;
+
+    case 'empty':
+      // Check if the promoted view exists and still has the block display.
+      $view = views_get_view('promoted');
+      if ($view && $view->access('block_1')) {
+        $nids = views_get_view_result('promoted', 'block_1');
```

Comment:
I've confirmed that Views will indeed execute the view twice, because `views_get_view()` returns a fresh unexecuted version of the view every time you call it.